book/standard_library.md

```diff
@@ -14,32 +18,218 @@ The standard library currently includes:
+
+There are several forms that can be used:
+
+#### 1. Import the submodule
+
+Examples:
+
+```nu
+use std/iter
+[2 5 "4" 7] | iter filter-map {|it| $it ** 2}
+
+use std/help
+help ls
+help commands
+```
+
+This form requires that you prefix the command using the submodule name. This can be useful in two scenarios.
+
+1. In the `use std/iter` example, it's important to prefix the commands with the `iter` namespace. If, on the other hand, you were to `use std/iter *`, you would shadow the built-in `find` command with the `iter` module's `find` command.
+
+2. In the `use std/help` example, `help` is both the submodule name and the name of the main command it exports. Using `use std/help` allows you to access:
+
+   ```nu
+   help
+   help aliases
+   help modules
+   help commands
+   help operators
+   help externs
+   ```
+
+   In this case, the `std/help` commands are meant to shadow (replace) the built-in versions.
+
+Submodules that are normally imported with `use std/<submodule>`:
+
+- `use std/assert`: `assert` and its subcommands
+- `use std/bench`: The benchmarking command `bench`
+- `use std/dirs`: The directory stack command `dirs` and its subcommands
+- `use std/input`: The `input display` command
+- `use std/help`: An alternative version of the `help` command and its subcommands which supports completion and other features
+- `use std/iters`: Additional `iters`-prefixed iteration commands. Note: See-also alternative option #3 below.
+- `use std/log`: The `log <subcommands>` such as `log warning <msg>`
+- `use std/math`: Mathematical constants such as `$math.E`. These can also be imported without a prefix using Form #2 below.
+
+#### 2. Import all submodule commands without a prefix
+
+For certain submodules, you will want the commands from a submodule to be available in the current namespace, so that you _can_ simply access the command by name. For instance:
+
+```nu
+use std/formats *
+ls | to jsonl
+```
+
+Submodules that are normally imported with `use std/<submodule> *`:
+
+- `use std/dt *`: Additional commands for working with `date` values
+- `use std/formats *`: Additional `to` and `from` format conversions
+- `use std/math *`: The math constants without a prefix, such as `$E`. Note that the prefixed form #1 above is likely more understandable when reading and maintaining code.
+- `use std/xml *`: Additional commands for working with XML data
+
+#### 3. Importing specific subcommands
+
+As with most modules, you can choose to import only a subset of the commands. For instance, the following would import the `zip-with` command without requiring that it be called with`iter zip-with`.
+
+```nu
+use std/iter [ zip-with ]
+```
+
+#### 4. Not Recommended - `use std <submodule>` and `use std <submodule> *`
+
+While it is _possible_ to import Standard Library submodules using a space-separated form:
+
+```nu
+use std log
+use std formats *
+```
+
+This is NOT RECOMMENDED. Using this form has the same performance impact as:
```

Comment:
I would just mention that “similar to `use std`, calling `use std <submodule>` first loads the ENTIRE `std`, then imports only the `<submodule>` from it. In contrast, `use std/<submodule>` loads ONLY the `<submodule>`”.

Reply:
Updated - Thanks!